crates/net/network/src/eth_requests.rs

```diff
@@ -216,10 +207,14 @@ where
                         .collect::<Vec<_>>(),
                 );
 
-                total_bytes += APPROX_RECEIPT_SIZE;
-
-                if total_bytes > SOFT_RESPONSE_LIMIT {
-                    break
+                match receipts.encode_max(SOFT_RESPONSE_LIMIT) {
+                    Ok(_) => {
+                        // If encode_max succeeds, continue accumulating receipts
+                    }
+                    Err(_) => {
+                        // If encode_max fails,stop the loop
+                        break;
+                    }
```

Comment:
if we want to keep approximations then we don't need this here, otherwise this would just be more additional work.